### DISCUSSION TITLE
Issues with wifi and physical device

### DISCUSSION THREAD

**Discussion by gorcsgergely** [firebase/flutterfire#8123] [Q&A]
Firebase authentication and firestore database operations are not working on Android, but only when a phone is using wifi.
As soon as wifi turned off, works perfectly with mobile data. Tested on multiple devices, multiple networks.
Emulator works perfectly with the same wifi. I don't really know how anybody is able to release any Flutter application using Firebase with that bug.

Did anybody manage to solve it? I see that it is a 2 year old problem.

The same issue is mentioned in numerous places by many people. Also there are closed issues here without any resolution.
https://github.com/FirebaseExtended/flutterfire/issues/2753
https://stackoverflow.com/questions/62349914/firebase-firestore-is-failing-over-wifi-on-physical-device-with-error-could-not 

Used dependecies in pubspec.yaml:
  firebase_core: ^1.12.0  
  cloud_firestore: ^3.1.8
  firebase_auth: ^3.3.7

**Comment by yamauchieduardo** [comment]
I notice this problem a long time ago like YOU (around 2 years). To solve that I build a whole architecture that start to listen a bunch of document when detect if the Firestore returned error code like

`PlatformException(Error performing get, Failed to get document because the client is offline., null),`

This is sad, because this issue happen a long time, and the worst part is: This is very critical to the users because this issue usually appear when the users are in the most critical part of the applications, in the on-boarding (login).
